alacritty_terminal/src/term/mod.rs

```diff
@@ -1050,17 +972,16 @@ impl<T> Term<T> {
         self.vi_mode_cursor.point.col = min(self.vi_mode_cursor.point.col, num_cols - 1);
         self.vi_mode_cursor.point.line = min(self.vi_mode_cursor.point.line, num_lines - 1);
 
-        // Recreate tabs list.
-        self.tabs.resize(self.grid.num_cols());
+        // Reset scrolling region.
+        self.scroll_region = Line(0)..self.num_lines();
 
-        // Reset scrolling region and selection.
-        self.scroll_region = Line(0)..self.grid.num_lines();
-        self.selection = None;
-    }
+        // Invalidate selection and tabs only when necessary.
+        if old_cols != num_cols {
+            self.selection = None;
 
-    #[inline]
-    pub fn mode(&self) -> &TermMode {
-        &self.mode
+            // Recreate tabs list.
+            self.tabs.resize(self.num_cols());
```

Comment:
I guess here you can use clear PR log to perform invalidation properly? If the goal is to keep selection after exiting search, you may want to compare new and old grid len, I guess it should work when columns are the same?

Reply:
I don't see what you mean by this, how would we use the clear PR to perform invalidation "properly"?

There is no invalidation going on when resizing, the content just moves. We cannot use invalidation on column change because the columns reflow and we do not know where from or where to and we do not need any kind of invalidation for line changes.

Reply:
I mean that in some cases when you're changing the amount of visible lines, you may want to invalidate selectiion? Not the other way around. I'm talking about properly invalidating, when we have the same amount of columns before and after, but a bit different lines_total value, since sometimes you may wan to clear selection here.

Reply:
> I mean that in some cases when you're changing the amount of visible lines, you may want to invalidate selectiion? 

Why would you ever want to do that? You're only changing the number of invisible lines, clearing any lines.

There is no range that is being invalidated.

Reply:
I mean than I don't really get an issue with this logic and how it's faulty? The only situation where I can think you may want to clear selection on resize, if the amount of lines was changed to a lower value, meaning that you must invalidate selection, knowing that you scrolled away a bunch of them, when your selection spans the entire grid. So technically selection became invalid, and so you could likely want to reset it as well.

In general, clear PR won't help much here, I guess. So that's why I've started talking about just invalidation. In general, selection should be `None` if it scrolled offscreen, so not in grid range anymore, otherwise I can expand from scrolled away selection, which is strange.